google/api_core/universe.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helpers for universe domain."""
+
+from google.auth.exceptions import MutualTLSChannelError
+from typing import Any, Optional
+
+DEFAULT_UNIVERSE = "googleapis.com"
+
+mTLS_Universe_Error = MutualTLSChannelError(
+    f"mTLS is not supported in any universe other than {DEFAULT_UNIVERSE}."
+)
+
+
+class _UniverseMismatchError(ValueError):
```

Comment:
Why keep this private? If we make it public, the user can check whether exceptions they get are of this type.